homeassistant/components/ozw/services.py

```diff
@@ -37,6 +44,103 @@ def async_register(self):
+            schema=vol.Schema(
+                {
+                    vol.Optional(const.ATTR_INSTANCE_ID, default=1): vol.Coerce(int),
+                    vol.Required(const.ATTR_NODE_ID): vol.Coerce(int),
+                    vol.Required(const.ATTR_CODE_SLOT): vol.Coerce(int),
+                }
+            ),
+        )
+
+        self._hass.services.async_register(
+            LOCK_DOMAIN,
+            const.SERVICE_CLEAR_USERCODE,
+            self.clear_usercode,
+            schema=vol.Schema(
+                {
+                    vol.Optional(const.ATTR_INSTANCE_ID, default=1): vol.Coerce(int),
+                    vol.Required(const.ATTR_NODE_ID): vol.Coerce(int),
+                    vol.Required(const.ATTR_CODE_SLOT): vol.Coerce(int),
+                }
+            ),
+        )
+
+    def set_usercode(self, service):
+        """Set the usercode to index X on the lock."""
+        instance_id = service.data[const.ATTR_INSTANCE_ID]
+        node_id = service.data[const.ATTR_NODE_ID]
+        code_slot = service.data[const.ATTR_CODE_SLOT]
+        usercode = service.data[const.ATTR_USERCODE]
+
+        lock_node = self._manager.get_instance(instance_id).get_node(node_id).values()
+
+        for value in lock_node:
+            if (
+                value.command_class == CommandClass.USER_CODE
+                and value.index == code_slot
+            ):
+                if len(str(usercode)) < 4:
+                    _LOGGER.error(
+                        "Invalid code provided: (%s) user code must be at least 4 digits",
+                        usercode,
+                    )
+                    break
+                value.send_value(usercode)
+                break
+
+    def get_usercode(self, service):
+        """Get a usercode at index X on the lock."""
+        instance_id = service.data[const.ATTR_INSTANCE_ID]
+        node_id = service.data[const.ATTR_NODE_ID]
+        code_slot = service.data[const.ATTR_CODE_SLOT]
+
+        lock_node = self._manager.get_instance(instance_id).get_node(node_id).values()
+
+        for value in lock_node:
+            if (
+                value.command_class == CommandClass.USER_CODE
+                and value.index == code_slot
+            ):
+                _LOGGER.info("User code at slot %s is: %s", code_slot, value.value)
+                break
+
+    def clear_usercode(self, service):
+        """Set usercode to slot X on the lock."""
+        instance_id = service.data[const.ATTR_INSTANCE_ID]
+        node_id = service.data[const.ATTR_NODE_ID]
+        code_slot = service.data[const.ATTR_CODE_SLOT]
+
+        lock_node = self._manager.get_instance(instance_id).get_node(node_id).values()
+
+        for value in lock_node:
+            if (
+                value.command_class == CommandClass.USER_CODE
+                and value.label == "Remove User Code"
+            ):
+                value.send_value(code_slot)
+                value.send_value(code_slot)
```

Comment:
Sending it twice ?

Reply:
Yes, for some reason I found sending it once didn't get it to take effect.